iothub/service/src/Common/IOThreadScheduler.cs

```diff
@@ -595,9 +595,13 @@ public void DebugVerifyEmpty()
         // by the GC anyway.
         [Fx.Tag.SecurityNote(Critical = "manages NativeOverlapped instance, can be called outside user context")]
         [SecurityCritical]
+#if NET451
         unsafe class ScheduledOverlapped
+#else
+        class ScheduledOverlapped
```

Comment:
Same as above - class contains unsafe code. It should be marked as `unsafe`. It would be ideal if we can remove it from ns builds.

Reply:
I have limited the unsafe code to net451 and hence changing the class to unsafe only for net451.

Reply:
The nature of the class (performing IOCP / Overlapped I/O) is unsafe. If all unsafe references have been removed then the class should not be doing anything.